packages/cozy-pouch-link/src/CozyPouchLink.js

```diff
@@ -539,6 +539,19 @@ class PouchLink extends CozyLink {
 
     if (!indexedFields) {
       indexedFields = getIndexFields(options)
+    } else if (partialFilter) {
+      // Some pouch adapters does not support partialIndex, e.g. with websql in react-native
+      // Therefore, we need to force the indexing the partialIndex fields to ensure they will be
+      // included in the actual index. Thanks to this, docs with missing fields will be excluded
+      // from the index.
+      // Note the $exists: false case should be handled in-memory.
+      indexedFields = Array.from(
+        new Set([...indexedFields, ...Object.keys(partialFilter)])
+      )
+      // FIXME: should properly handle n-level attributes
```

Comment:
What's the impact for now? And should we handle this soon? Or it is for an hypothetical future?

Reply:
For now it means that if we have queries with partialIndex having attributes not on the root level, i.e. with `$and: [{...}]`, we won't benefit from this fix. Thus slow queries. 
So *hopefully* there is no regression  regarding the current offline, **but** we should test it carefully